src/engine/x86-64/X86_64Interpreter.v3

```diff
@@ -3700,8 +3648,64 @@ class X86_64InterpreterGen(ic: X86_64InterpreterCode, w: DataWriter) {
 		fatal("ran out of buffer space");
 		return w;
 	}
+	// generates {resume} code (factored out for both {resume} and {resume_throw})
+	// params:
+	// - r_cont_id[input]: index of the continuation type
+	// - r_tag_id[input]: index of the {resume_throw} tag
+	// - r_cont[output]: pointer to the popped continuation
+	private def genResume(r_cont_id: X86_64Gpr, r_tag_id: X86_64Gpr, r_cont: X86_64Gpr, tmp1: X86_64Gpr, is_resume_throw: bool) {
```

Comment:
The `is_resume_throw` parameter makes me think the split could be improved a little. Maybe we can factor out the first part (before the `if`) and then inline the rest in the call sites.